frameworks/keyed/svelte/src/Main.html

```diff
@@ -72,11 +70,19 @@ <h1>Svelte (keyed)</h1>
       data = buildData(10000);
       selected = undefined;
     },
-    select = (id) => selected = id,
+    select = id => selected = id,
     swapRows = () => {
       if (data.length > 998) {
         data = [data[0], data[998], ...data.slice(2, 998), data[1], data[999]];
       }
+    },
+    click = ({ target }) => {
+      const id = Number.parseInt(target.closest("tr").firstChild.firstChild.nodeValue);
```

Comment:
@ryansolid

so, we're still doing this, huh?

Reply:
Fair is fair? I added implicit Event Delegation to my libraries to align with what the VDOM implementations. And I like the story it tells for code aesthetics ("Write less code"), but it is hardly a fair comparison when implementations aren't allowed to use the exact same mechanisms.

I realize the same could be argued about any trick VanillaJS does, but it just means the floor is pretty low. It's the same old discussion again. Which for me comes down to how does the implementation want to present itself. As long as the minimum requirements are met (Ie.. completely data driven, ie renderable from non-DOM state at any given point) it is up to them how nasty and complicated they want to make their implementation.

So yes I was hesitant to do this initially. Especially without @Rich-Harris approval since I get the impression Svelte is a library that cares more about aesthetics than performance. But absent of that I'd say it's probably fair to align this with other high performing non-VDOM implementations like Surplus, LitHTML, LighterHTML, and so on.

As always ready to open this debate again. But I think it is very difficult for any implementation/implementor that cares about performance and doesn't include implicit event delegation in their libraries for various reasons, to even understand why this is an issue in itself. I implemented implicit event delegation and I still don't.